### DISCUSSION TITLE
Integrating Bazel with Kurtosis

### DISCUSSION THREAD

**Discussion by shreyasbhat0** [kurtosis-tech/kurtosis#747] [Ideas]
Hey Everyone .

I had a thought on using Kurtosis for both Build and Deployment , to make that happen if bazel could be integrated with kurtosis ,where bazel builds can be used via kurtosis starlark package to execute it . And the generated binary / packages / docker images can be used later in Kurtosis packages for deployment . 

Example :
```
def build_some_binary(plan,args):
     binary_file =  plan.build(
         src ="src_files",
        out = "file_name",
        deps = "any deps"
      )
     service_config = ServiceConfig(
      files = {
        FILE_PATH : binary_file
        }
     )
```

**Comment by leeederek** [comment]
Thanks @shreyasbhat0 for this submission! Just to confirm my understanding: you're suggesting a `build binary` functionality in Kurtosis, and are proposing Bazel be the tool we use under the hood to build those binaries?

**Reply by shreyasbhat0** [reply]
yeah exactly
